src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -505,6 +528,76 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     }
   }
 
+  override def checkGoogleGroupExists(groupEmail: String): Boolean = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    val checkGroupRequest = directoryService.groups.get(groupEmail)
+    Try(executeGoogleRequest(checkGroupRequest)) match {
+      case Failure(_) => false
+      case Success(_) => true
+    }
+  }
+
+  override def deleteGoogleGroup(groupEmail: String): Unit = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    val deleteGroupRequest = directoryService.groups.delete(groupEmail)
+    Try(executeGoogleRequest(deleteGroupRequest)) match {
+      case Failure(_) => {
+        logger.warn(s"Failed to delete group $groupEmail")
+      }
+      case Success(_) => {}
+    }
+  }
+
+  /**
+    * create a new Google group
+    * @param groupEmail     the name of the new Google group to be created. it must be formatted as an email address
+    * @return               Option of the groupEmail address if all above is successful, otherwise if the group creation
+    *                        step failed, return Option of an empty string
```

Comment:
Is the group email passed in the same thing that would be returned on success? If so, then this method could also just return a boolean. If they're different, then yes, this makes sense.

Reply:
I wanted to make sure we get the name of the group that's created, in case for some unimaginable reason it would change. I can't imagine why it would, so it probably would be safe to return a Boolean. But that would change the logic of our unit tests and since it works now I'd rather keep it this way unless it seems problematic @rushtong

Reply:
👍 Sounds good to me, thanks for thinking that out!